### DISCUSSION TITLE
Looping through all the slides stopped working for me

### DISCUSSION THREAD

**Discussion by darrenhundt** [nolimits4web/swiper#6331] [Q&A]
I have been using swiper for a client's site for a while. 
I was using some data attributes of the slides to control some things so i would often loop through the slides. 
I'm not a javascript expert and was using jquery for the slider.
Today we noticed that looping through the slides using
`swiper.slides.each(function(){
    // I would use $(this) to reference each slide
});`
stopped working. To get it to work now it needs to be
`swiper.slides.forEach(function(e,i){
    // now referencing each slide using i instead of $(this)
});`
As far as I know, nobody changed anything in our code, so i was wondering what we could have done that made it stop working?

**Comment by dugweb** [comment]
Are you linking to the swiper js files through their CDN? I had a similar issue happen. No code changes on the site, but the [sliders stopped working](https://github.com/nolimits4web/swiper/discussions/6320) and needed to have the website's code updated to restore functionality. I have to believe that the CDN's code was updated.

**Reply by darrenhundt** [reply]
Yes. I was using the CDN. I have since downloaded my own copy of the file and made changes to my code. Thanks for the tip.
